docs/docs/en/guides/features/projects/best-practices.md

```diff
@@ -63,3 +63,29 @@ let target = Target(
 ```
 <!-- -->
 :::
+
+### Dependencies
+
+#### Force resolved versions on CI
+
+When installing Swift Package Manager dependencies on CI, we recommend using the `--force-resolved-versions` flag to ensure deterministic builds:
+
+```bash
+tuist install --force-resolved-versions
+```
+
+This flag ensures that dependencies are resolved using the exact versions pinned in `Package.resolved`, eliminating issues caused by non-determinism in dependency resolution. This is particularly important on CI where reproducible builds are critical.
+
+If you want this behavior to be the default for all environments (including developer machines), you can configure it in your `Tuist.swift`:
+
+```swift
+import ProjectDescription
+
+let tuist = Tuist(
+  project: .tuist(
+    installOptions: .options(
+        passthroughSwiftPackageManagerArguments: ["--force-resolved-versions"]
+    )
+  )
+)
+```
```

Comment:
I don't recommend setting this in `installOptions` as adding new packages will become painful.